### PR TITLE
feat: added app config management

### LINKED ISSUE #11: [FEATURE] Allow port to be customised at runtime
**Is your feature request related to a problem? Please describe.**
The port is currently hardcoded. While we've chosen a rather unique port number, there's always chance of conflict.

**Describe the solution you'd like**
Allow the port number to be customised via an environment variable, CLI flag, or both.

### PR DESCRIPTION
**Pull request by ashiquem**
Config management using yml files
Added port defaults

### CODE REVIEW

File: internal/api/config.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 package api
 
 type Config struct {
-	Port string
```
Comment (disposedtrolley) [inline internal/api/config.go:4]: Could we include an `env:"PORT"` tag here too? I think we'll supply much of the config in prod via environment variables.

File: config.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+port: 1927
```
Comment (disposedtrolley) [inline config.yml:1]: This is going to be fantastic for local dev!

**Review by disposedtrolley — COMMENTED** [review]
Nice work! I like the dual approach of configuration via env vars and files that `cleanenv` afford us.

What are your thoughts on renaming the config file to `config-dev.yml` and using the YAML method specifically for dev and test (we could have another `config-test.yml` when we need it)?

The production config will be much easier to manage via environment variables (you can see an example of some Terraform config here: https://github.com/omshub/core-api/pull/14/files#diff-d7b09c72489ecbd047bf9954f1f51ec38e32bbca8de05352e0dfc4537657e150R43-R54)

File: internal/api/config.go (hunk 1 above)
Comment (ashiquem) [reply internal/api/config.go:4]: Sure. Do we want to use a more specific name such as `SERVER_PORT` for the env variable?

**Comment by ashiquem** [comment]
> Nice work! I like the dual approach of configuration via env vars and files that `cleanenv` afford us.
> 
> What are your thoughts on renaming the config file to `config-dev.yml` and using the YAML method specifically for dev and test (we could have another `config-test.yml` when we need it)?
> 
> The production config will be much easier to manage via environment variables (you can see an example of some Terraform config here: https://github.com/omshub/core-api/pull/14/files#diff-d7b09c72489ecbd047bf9954f1f51ec38e32bbca8de05352e0dfc4537657e150R43-R54)


I like the idea of having environment-specific configurations. The env variable setups in the linked PR looks great. Is it possible to keep the environment-specific configurations to the infra code instead of doing that here in the app code?

File: internal/api/config.go (hunk 1 above)
Comment (disposedtrolley) [reply internal/api/config.go:4]: I prefer `SERVER_PORT`. We'll need to modify the [Terraform config](https://github.com/omshub/core-api/blob/main/ops/main.tf#L31) to add `SERVER_PORT` as an env var (DigitalOcean was automatically mapping `http_port` to `PORT` for us).

**Comment by disposedtrolley** [comment]
> Is it possible to keep the environment-specific configurations to the infra code instead of doing that here in the app code?

Yeah absolutely! I used `config-test.yml` as an example of configuration we might apply when running unit/integration tests, rather than a "test" environment.

Edit: actually that would require the filename to be customisable, which it isn't currently. We could allow the config to be specified in a CLI flag using the [flag package](https://pkg.go.dev/flag) or [Cobra](https://github.com/spf13/cobra), but that might be best served by another PR.

**Review by awpala — APPROVED** [review]

**Review by disposedtrolley — APPROVED** [review]

**Pull request merged**